### DISCUSSION TITLE
Render function with N-space - what am I doing wrong here to pass children?

### DISCUSSION THREAD

**Discussion by mitya33** [tusen-ai/naive-ui#6232] [Q&A]
As far as I know this is valid use of Vue's `h()` function:

```
h(NSpace, {size: 'small'}, [
	h('span', {textContent: 'foo'}),
	h('span', {textContent: 'bar'}),
])
```

Yet this throws the warning:

> _[Vue warn]: Non-function value encountered for default slot. Prefer function slots for better performance. _

What would be correct here? I've tried:

```
h(NSpace, {size: 'small'}, [
	() => h('span', {textContent: 'foo'}),
	() => h('span', {textContent: 'bar'}),
])
```

...but that no joy.

**Comment by jahnli** [comment]
This issue does not have any recent activity. If you are still experiencing similar problems, open a new error, including a minimal copy of the problem

**Comment by jahnli** [comment]
This issue does not have any recent activity. If you are still experiencing similar problems, open a new error, including a minimal copy of the problem
